lib/galaxy/model/__init__.py

```diff
@@ -2073,13 +2073,15 @@ def get_converted_dataset(self, trans, target_ext):
             raise NoConverterException("A dependency (%s) is missing a converter." % dependency)
         except KeyError:
             pass  # No deps
-        new_dataset = next(iter(self.datatype.convert_dataset( trans, self, target_ext, return_output=True, visible=False, deps=deps, set_output_history=True ).values()))
+        new_dataset = next(iter(self.datatype.convert_dataset( trans, self, target_ext, return_output=True, visible=False, deps=deps, set_output_history=True, target_context=target_context ).values()))
+        new_dataset.hid = self.hid
```

Comment:
Thanks a lot for reporting this. I am working on a potential solution here: https://github.com/galaxyproject/galaxy/pull/3149.